src/event_pusher/mod_event_pusher_push_plugin_defaults.erl

```diff
@@ -30,15 +30,15 @@
 %% Callback 'should_publish'
 -spec should_publish(From :: jid:jid(), To :: jid:jid(), Packet :: exml:element()) ->
                             boolean().
-should_publish(_From, To = #jid{luser = LUser, lserver = LServer}, _Packet) ->
+should_publish(_From, To = #jid{luser = LUser, lserver = LServer}, Packet) ->
     try ejabberd_users:does_user_exist(LUser, LServer) of
         false ->
             false;
         true ->
-            ejabberd_sm:is_offline(To)
+            ejabberd_sm:is_offline(To) andalso has_body(Packet)
     catch
         _:_ ->
-            ejabberd_sm:is_offline(To)
+            ejabberd_sm:is_offline(To) andalso has_body(Packet)
```

Comment:
Good catch. In https://github.com/esl/MongooseIM/blob/25b0d1d6bf6342918e02abfac881fe6cde202713/src/event_pusher/mod_event_pusher_push.erl#L115 we only checked if the `<body>` exists, but not if it is empty.